### PR TITLE
Updating dependencies

### LINKED ISSUE #2: Upgrade kotlin to 1.4.x
Kotlin dependency is currently at 1.3.72.  Upgrade it to at least 1.4.10

### LINKED ISSUE #5: Upgrade Android Gradle plugin to 4.2.0
Android Gradle plugin is currently at 3.6.4.
Upgrade it to at least 4.2.0.

### LINKED ISSUE #6: Upgrade Gradle to 6.6.1
Currently Gradle is at 5.6.4

Upgrade it to at least 6.6.1

### PR DESCRIPTION
**Pull request by CWJamieson**
Addresses:
- [VSL-2](https://github.com/textnow/vessel/issues/2) Upgrade Kotlin version from 1.3.72 ➞ 1.5.21
- [VSL-5](https://github.com/textnow/vessel/issues/5) Upgrade Android Gradle plugin version from 3.6.4 ➞ 7.0.2
- [VSL-6](https://github.com/textnow/vessel/issues/6) To match that, upgraded Gradle from 5.6.4 ➞ 7.0.2
- This in turn required changing our JDK version from 8 ➞ 11
- Upgrade the build tools from 30.0.2 ➞ 31.0.0
- Upgrade Room from 2.2.5 ➞ 2.4.0-alpha04 to be compatible with the Kotlin upgrade (2.3.0 failed to compile)
- Upgrade jacoco from 0.8.6 ➞ 0.8.7
- Upgrade Koin from 2.1.6 ➞ 3.1.2
- Updated build tools from 30.0.2 ➞ 31.0.0
- Updated compile SDK from 30 ➞ 31

### CODE REVIEW

File: buildSrc/src/main/kotlin/Versions.kt [hunk 1]
```diff
@@ -22,18 +22,18 @@
  * SOFTWARE.
  */
 object Versions {
-    const val kotlin = "1.3.72"
-    const val androidGradle = "3.6.4"
+    const val kotlin = "1.5.21"
+    const val androidGradle = "7.0.1"
```
Comment (malachid) [inline buildSrc/src/main/kotlin/Versions.kt]: you should bump this to 7.0.2 to match the gradle version

File: vessel-runtime/build.gradle.kts [hunk 2]
```diff
@@ -7,22 +7,18 @@ plugins {
 }
 
 android {
-    compileSdkVersion(30)
-    buildToolsVersion = "30.0.2"
+    compileSdk = 31
```
Comment (malachid) [inline vessel-runtime/build.gradle.kts:10]: make sure the changelog reflects this change

File: buildSrc/src/main/kotlin/Versions.kt (hunk 1 above)
Comment (CWJamieson) [reply buildSrc/src/main/kotlin/Versions.kt]: AGP is only matching Gradle's first two version numbers, so there is no 7.0.2 and 7.0.1 is just a minor bug hotfix, unrelated to Gradle's 7.0.1/2 https://developer.android.com/studio/releases/gradle-plugin#7-0-0

File: vessel-runtime/build.gradle.kts (hunk 2 above)
Comment (CWJamieson) [reply vessel-runtime/build.gradle.kts:10]: Will do!

**Review by carterhudson — APPROVED** [review]

File: buildSrc/src/main/kotlin/Versions.kt (hunk 1 above)
Comment (malachid) [reply buildSrc/src/main/kotlin/Versions.kt]: Unfortunately, their documentation is out of date
https://mvnrepository.com/artifact/com.android.tools.build/gradle/7.0.2

Comment (CWJamieson) [reply buildSrc/src/main/kotlin/Versions.kt]: Oh I see, there was an update for AGP that hasn't been posted on the releases doc yet, updated!

**Review by malachid — APPROVED** [review]
Looks good. Tests passed.

**Pull request merged**